File: problem/Multiply_Strings/multiply.py
class Solution:
    def multiply(self, num1, num2):
        n1, n2 = len(num1), len(num2)
        num = [0] * (n1 + n2)

        for i in range(n1 - 1, -1, -1):
            for j in range(n2 - 1, -1, -1):
                mul = (ord(num1[i]) - ord('0')) * (ord(num2[j]) - ord('0'))
                sum_ = num[i + j + 1] + mul
                num[i + j + 1] = sum_ % 10
                num[i + j] += sum_ // 10

        i = 0
        while i < len(num) - 1 and num[i] == 0:
            i += 1

        return ''.join(str(d) for d in num[i:])

File: problem/Multiply_Strings/multiply.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
function multiply(num1, num2) {
    let n1 = num1.length, n2 = num2.length;
    let num = new Array(n1 + n2).fill(0);

    for (let i = n1 -1; i >= 0; i--) {
        for (let j = n2 -1; j >= 0; j--) {
            let mul = (num1.charCodeAt(i) - 48) * (num2.charCodeAt(j) - 48);
            let sum = num[i + j + 1] + mul;
            num[i + j + 1] = sum % 10;
            num[i + j] += Math.floor(sum / 10);
        }
    }

    while (num.length > 1 && num[0] === 0) {
        num.shift();
    }

    return num.join('');
}
